js/src/vm/PortableBaselineInterpret.cpp

```diff
@@ -1828,6 +1834,16 @@ ICInterpretOps(BaselineFrame* frame, VMFrameManager& frameMgr, State& state,
     DISPATCH_CACHEOP();
   }
 
+  CACHEOP_CASE(MetaScriptedThisShape) {
+    uint32_t thisShapeOffset = icregs.cacheIRReader.stubOffset();
+    Shape* thisShape = reinterpret_cast<Shape*>(
+        cstub->stubInfo()->getStubRawWord(cstub, thisShapeOffset));
+    // This op is only metadata for the Warp Transpiler and should be ignored.
+    (void)thisShape;
```

Comment:
Can we remove the `thisShape` fetch? I know that it should be optimized away here, but I'd prefer not to write it in the first place (for several reasons: to avoid the overhead in debug builds, to avoid the clutter when reading this code, ...)

Reply:
Oh, this is another case where I was trying to follow your lead. :laughing:

```
  CACHEOP_CASE(AssertPropertyLookup) {
    ObjOperandId objId = icregs.cacheIRReader.objOperandId();
    uint32_t idOffset = icregs.cacheIRReader.stubOffset();
    uint32_t slotOffset = icregs.cacheIRReader.stubOffset();
    // Debug-only assertion; we can ignore.
    (void)objId;
    (void)idOffset;
    (void)slotOffset;
    DISPATCH_CACHEOP();
  }
```

I can delete the fetches in both places if you prefer? I'd certainly rather have less code in these ops.

Aside from that, I need to dig into why some debug asserts are failing before we can merge this.

Reply:
Ah, those are different: we do actually need to call those methods on the reader to advance the offset. What I meant here was the call to `getStubRawWord` -- that shouldn't be needed here, I think?